File: c/2874.c
#define max(a, b) ((a > b) ? a : b)

long long maximumTripletValue(int* nums, int n) {
    long long res = 0, diff = 0;
    int i = 0;
    for (int j = 0; j < n; j++) {
        res = max(res, diff * nums[j]);
        diff = max(diff, i - nums[j]);
        if (nums[j] > i)
            i = nums[j];
    }
    return res;
}

File: javascript/2874.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumTripletValue = function (nums) {
    var res = 0, diff = 0, i = 0;

    var max = function (a, b) { return a > b ? a : b; }

    for (let j of nums) {
        res = max(res, diff * j);
        diff = max(diff, i - j);
        i = max(i, j);
    }
    return res;
};
